src/main/java/gg/uhc/uhc/modules/health/PercentHealthObjectiveModule.java

```diff
@@ -211,6 +222,27 @@ public void on(PlayerQuitEvent event) {
         trackedHealth.remove(event.getPlayer().getUniqueId());
     }
 
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void on(EntityDamageEvent event) {
+        Entity entity = event.getEntity();
+        if (!(entity instanceof Player)) {
+            return;
+        }
+
+        Player player = (Player) entity;
+        // Calculate new health based on final damage instead of waiting for one tick
```

Comment:
If we wait one tick here and remove ignoreCancelled=true we can take into account damage events that cancel and set the health manually